.github/workflows/magma-build-3rd-party.yml

```diff
@@ -0,0 +1,99 @@
+# Copyright 2022 The Magma Authors.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Magma Build & Publish 3rd Party Dependencies
+on:
+  push:
+    branches:
+      - master
+      - 'v1.*'
+    paths:
+      - third_party/**
+      - .github/workflows/magma-build-3rd-party.yml
+  pull_request:
+    types: [ opened, reopened, synchronize ]
+    branches:
+      - master
+      - 'v1.*'
+    paths:
+      - third_party/**
+      - .github/workflows/magma-build-3rd-party.yml
+  schedule:
+    - cron: 0 4 * * 0
+  workflow_dispatch:
+    inputs:
```

Comment:
I think the most common use-case is that only one library is updated. Is the idea here that always all libraries are re-build and uploaded?
Is the jfrog cli robust vs uploading an already existing artifact?

Reply:
>  Is the jfrog cli robust vs uploading an already existing artifact?

Yes, it is. Already existing packages will be overwritten.

> Is the idea here that always all libraries are re-build and uploaded?

Yes.
- Reuploading existing packages build from same source does is possible and does not hurt.
- Frequently rebuilding everything allows for better monitoring if all dependencies are still in a "healthy" state.
- Frequently uploading everything tests also the complete publishing workflow (including the upload, credentials, etc.) end-to-end, which gives additional value in terms of certainty of deliverability of that components.
- I see an advantage to always have the complete set in the current aligned, as this is much easier to track. I.e. it is clear where the artifacts come from and what state the are / were in. This should be homogeneous across all artifacts than.
- As above: It is just cheaper and more resilient to have this simple and less complex solution, compared to a sophisticated logic here. I do not see any benefit of the additional effort.

Reply:
The acceptance criteria of the ticket say

> the workflow is executed regularly and builds all artifacts without uploading them - workflow fails if one of the artifact builds fail

The way I understand this is that the idea is not to overwrite the uploaded artifacts, only to verify that the build still works.

Reply:
> I see an advantage to always have the complete set in the current aligned, as this is much easier to track. I.e. it is clear where the artifacts come from and what state the are / were in. This should be homogeneous across all artifacts than.

i don't agree that overwriting existing files makes tracking easier. On the contrary, if we have immutable artifacts we can make statements like _"last month we were building our magma package with nettle version 1.2.3-0. Since the start of this month, we build it with nettle version 1.2.3-1, and we see the following issue..."_. When overwriting the versions we cannot make such statements anymore because we just don't know.

Reply:
Only manual deployments than.